### PR TITLE
[Az.Resources] Enable role assignment when subcription is not provided

### PR DESCRIPTION
**Pull request by NoriZC**
## Description

Scope validation parameter is changed because when defined [ValidateScope](https://github.com/Azure/azure-powershell/blob/c06d9e22aa144cb1f737d8a2688bda24ecfc026a/src/Resources/Resources/Models.Authorization/AuthorizationClient.cs#L668), we only consider `string.IsNullOrEmpty(scope)` together with `AllowEmpty`, in this case if `!allowEmpty && string.IsNullOrEmpty(scope)` case is added before that, there will be breaking changes. 

Meanwhile by setting `AllowEmpty = true`, user will receive accurate error message  directly from service side. 
## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Resources/Resources/RoleAssignments/NewAzureRoleAssignmentCommand.cs [hunk 1]
```diff
@@ -323,7 +323,7 @@ public override void ExecuteCmdlet()
                     ResourceGroupName = ResourceGroupName,
                     ResourceName = ResourceName,
                     ResourceType = ResourceType,
-                    Subscription = DefaultProfile.DefaultContext.Subscription.Id,
+                    Subscription = DefaultProfile.DefaultContext.Subscription != null ? DefaultProfile.DefaultContext.Subscription.Id : "",
```
Comment (isra-fel) [inline src/Resources/Resources/RoleAssignments/NewAzureRoleAssignmentCommand.cs:326]: try this
```suggestion
                    Subscription = DefaultProfile.DefaultContext.Subscription ?? "",
```

**Review by isra-fel — APPROVED** [review]

**Review by isra-fel — COMMENTED** [review]
Please make the title more accurate and explain your code change either inline or in PR description. Line 326 is clear, but 334 isn't.
Thanks.

**Comment by NoriZC** [comment]
> Please make the title more accurate and explain your code change either inline or in PR description. Line 326 is clear, but 334 isn't. Thanks.

updated, please help check whether I can be understood

**Review by Nickcandy — APPROVED** [review]

**Pull request merged**